rest/route/middleware.go

```diff
@@ -914,6 +916,46 @@ func (m *EventLogPermissionsMiddleware) ServeHTTP(rw http.ResponseWriter, r *htt
 	next(rw, r)
 }
 
+// NewGithubAuthMiddleware returns a middleware that verifies the payload.
+func NewGithubAuthMiddleware() gimlet.Middleware {
+	return &githubAuthMiddleware{}
+}
+
+type githubAuthMiddleware struct{}
+
+func (m *githubAuthMiddleware) ServeHTTP(rw http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+	githubSecret := []byte(evergreen.GetEnvironment().Settings().Api.GithubWebhookSecret)
+
+	payload, err := github.ValidatePayload(r, githubSecret)
+	if err != nil {
+		grip.Error(message.WrapError(err, message.Fields{
+			"source":  "GitHub hook",
+			"message": "rejecting GitHub webhook",
+			"msg_id":  r.Header.Get("X-Github-Delivery"),
+			"event":   r.Header.Get("X-Github-Event"),
+		}))
+		gimlet.WriteResponse(rw, gimlet.MakeJSONErrorResponder(errors.Wrap(err, "validating GitHub payload")))
+		return
+	}
+
+	r = setPayload(r, payload)
+	next(rw, r)
+}
+
+func setPayload(r *http.Request, payload []byte) *http.Request {
```

Comment:
Since this is just a one-liner and used in one place I would avoid making a package-wide `setPayload` function since it pollutes the package namespace (or call it something more specific like `setGitHubPayload`).